Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +7,197 @@ public static class BalanceStats
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var firstAccount = peopleAndBalances[0].Split(", ");
+            var firstBalance = Lesson7.StringArrayToFloatArray(firstAccount[1..]);
+
+            var names = new [] { firstAccount[0] };
+            var highestBalance = firstBalance[^1];
+
+            for (int i = 1; i < peopleAndBalances.Length; i++)
+            {
+                var currentAccount = peopleAndBalances[i].Split(", ");
+                var currentBalance = Lesson7.StringArrayToFloatArray(currentAccount[1..]);
+
+                var currentName = currentAccount[0];
+                var currentHighest = currentBalance[^1];
+
+                // if same as others add it to the list
+                if (Lesson7.EqualFloats(currentHighest, highestBalance))
+                {
+                    names = Lesson7.AppendString(names, currentName);
+                }
+
+                if (currentHighest > highestBalance)
+                {
+                    names = new [] { currentName };
+                    highestBalance = currentHighest;
+                }
+            }
+
+            string formatedNames = Lesson7.FormatArrayToString(names);
+            
+            if (names.Length == 1)
+            {
+                return $"{formatedNames} is the richest person. ¤{highestBalance}.";
+            }
+            return $"{formatedNames} are the richest people. ¤{highestBalance}.";
         }
 
         /// <summary>
         /// Return name and current money of the most poor person.
         /// </summary>
         public static string FindMostPoorPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var firstAccount = peopleAndBalances[0].Split(", ");
+            var firstBalance = Lesson7.StringArrayToFloatArray(firstAccount[1..]);
+
+            var names = new string[] { firstAccount[0] };
+            var lowestBalance = firstBalance[^1];
+
+            for (int i = 1; i < peopleAndBalances.Length; i++)
+            {
+                var currentAccount = peopleAndBalances[i].Split(", ");
+                var currentBalance = Lesson7.StringArrayToFloatArray(currentAccount[1..]);
+
+                var currentName = currentAccount[0];
+                var currentLowest = currentBalance[^1];
+
+                // if same as others add it to the list
+                if (Lesson7.EqualFloats(currentLowest, lowestBalance))
+                {
+                    names = Lesson7.AppendString(names, currentName);
+                }
+
+                if (currentLowest < lowestBalance)
+                {
+                    names = new string[] { currentName };
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the array type; it is redundant.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5001581)